File: algo/length-of-last-word/lengthOfLastWord.go
func lengthOfLastWord(s string) int {
    x, ret := len(s) - 1, 0
    for x >= 0 && s[x] == ' ' {
        x--
    }
    for x >= 0 && s[x] != ' ' {
        x--
        ret++
    }
    return ret
}

File: algo/length-of-last-word/lengthOfLastWord.c
int lengthOfLastWord(char* s) {
    char* p;
    int length;
    
    if (s == NULL) {
        return 0;
    }
    
    p = s;
    length = 0;
    while (*p) {
        ++p;
    }
    --p;
    while (p >= s && *p == ' ') {
        --p;
    }
    while (p >= s && *p != ' ') {
        ++length;
        --p;
    }
    return length;
}